#4: Added parameters

developbarry opened the pull request.

Commit:
#4: Added parameters

Commit:
#4: Added parameters

gofix commented on example.html:
```diff
@@ -0,0 +1,345 @@
+
+  <body>
+    <section>
+      <h1>Large</h1>
```
This isn't what I had in mind. I'd like to suggest you create 3 examples (instead of 1 `example.html` with 3 different sized grids):
- example-15-columns-grid.html
- example-12-columns-grid.html
- example-4-columns-grid.html

... because that's exactly what's out there in real life sites!

Each example inludes its own css (i.e. `css/15-columns-grid.css`, `css/12-columns-grid.css`, `css/4-columns-grid.css`).

Commit:
4 example pages with 4 different kind of grids.

gofix commented on scss/example-15.scss:
```diff
@@ -0,0 +1,46 @@
+);
+
+
+.grid_15 {
```
al deze debug styles zou ik naar 1 centrale partial verhuizen. Eventueel maak je daarin een array van kleuren, die je dan met een for loop voor ieder van de relevante grid maten laat uitdelen :)

gofix commented on scss/example-12.scss:
```diff
@@ -0,0 +1,47 @@
+
+  
+
+.grid_15 {
```
Zie opmerking verderop over deze debug kleur stijlen.

gofix commented on example-4columns.html:
```diff
@@ -0,0 +1,212 @@
+                </div>
+              </div>
+              <div class="grid-blok grid_1">
+                  <div class="grid-element">
```
Deze indentering klopt niet.

Commit:
Erased block variable and maked some structure changes (html pages in a html folder)

gofix commented on scss/mixins/_viewport-logic.scss:
```diff
@@ -0,0 +1,176 @@
+$zone-sizes: $grid-zones;
+@each $zone-index in $zone-sizes {
+  .grid-zone.grid_#{$zone-index} {
+    @if $zone-index <= $current-columns {
```
Als deze `if` `false` is levert dit in de css een lege rule op. Je zult de `if` OM de hele rule willen zetten denk ik?

gofix commented on scss/mixins/_viewport-logic.scss:
```diff
@@ -0,0 +1,176 @@
+      @include span($zone-index of $current-columns no-gutter);
+    }
+  }
+  @if $zone-index > $current-columns {
```
Schreeuwt dit niet om een `@if` met een `@else` ?
http://sass-lang.com/documentation/file.SASS_REFERENCE.html#_if

gofix commented on scss/mixins/_viewport-logic.scss:
```diff
@@ -0,0 +1,176 @@
+        }
+      }
+    }
+    @if $container-size >= 960 {
```
oei, een harde maat? misschien bedoel je de situatie waarbij `$columns-large = $current-columns` ?

gofix commented on scss/mixins/_viewport-logic.scss:
```diff
@@ -0,0 +1,176 @@
+    }
+  }
+
+  @if $zone-index <= $current-columns {
```
volgens mij zouden we in eerste instantie al het pullen en pushen alleen doen bij `$columns-large = $current-columns`

gofix commented on scss/mixins/_viewport-logic.scss:
```diff
@@ -0,0 +1,176 @@
+  }
+}
+
+@if $current-columns == $columns-small and $current-columns != $columns-medium {
```
dat deel achter de `and` zal toch nooit waar zijn als het eerste deel waar is? of doel je op een configuratie waarin iemand `$columns-small` gelijkstelt aan `$columns-medium`

gofix commented on scss/mixins/_viewport-logic.scss:
```diff
@@ -0,0 +1,176 @@
+    .grid-zone.grid_#{$zone-size} {
+      @include span(3 of 3 no-gutter);
+
+      @for $blok-size from 4 through $zone-size {
```
deze 4 is een hard getal, dat moet niet, maar moet afhankelijk zijn van het de `$columns-small` denk ik?

gofix commented on scss/mixins/_viewport-logic.scss:
```diff
@@ -0,0 +1,176 @@
+        }
+      }
+
+      @for $blok-size from 1 through 3 {
```
deze 3 is een hard getal, dat willen we volgens mij niet

gofix commented on scss/mixins/_viewport-logic.scss:
```diff
@@ -0,0 +1,176 @@
+    }
+  }
+
+  $zone-sizes: 3;
```
harde getallen?

gofix commented on scss/mixins/_viewport-logic.scss:
```diff
@@ -0,0 +1,176 @@
+  $zone-sizes: 3;
+  @each $zone-index in $zone-sizes {
+    .grid-zone.grid_#{$zone-index} {
+      @include span(6 of 6 no-gutter);
```
harde getallen?

gofix commented on scss/mixins/_viewport-logic.scss:
```diff
@@ -0,0 +1,176 @@
+      }
+    }
+
+    @if $container-size < 320px {
```
ik wil geen harde getallen (`320px`) zien!

gofix commented on scss/mixins/_viewport-logic.scss:
```diff
@@ -0,0 +1,176 @@
+
+    @if $container-size < 320px {
+      @if $zone-index < 4 and $zone-index > 2 {
+        .grid-zone.grid_12 .grid-blok.grid_3 {
```
`grid_12` bevat een hard getal!

Commit:
Erased block variable and container size variable in the baseline default file.
I maked some structure changes (html pages in a html folder)
I maked another file called debugcolors for the debug color settings.

Commit:
Changed hard numbers into variables
Changed some if else statements

Commit:
Changed the logic
Added pug
Made pug pages

Commit:
Installed Pug
Added pug pages
Added no-gutter
